src/main/scala/io/iohk/ethereum/network/p2p/MessageDecoders.scala

```diff
@@ -38,45 +35,59 @@ object NetworkMessageDecoder extends MessageDecoder {
 // scalastyle:off
 object EthereumMessageDecoder extends MessageDecoder {
 
-  override def fromBytes(msgCode: Int, payload: Array[Byte], protocolVersion: Version): Message =
+  override def fromBytes(msgCode: Int, payload: Array[Byte], protocolVersion: Version): Message = {
+    import io.iohk.ethereum.network.p2p.messages.CommonMessages.NewBlock._
+    import io.iohk.ethereum.network.p2p.messages.CommonMessages.Status._
+
     (protocolVersion, msgCode) match {
       //wire protocol
       case (_, Hello.code) => payload.toHello
 
-      //FIXME: I still have an issue with protocolVersion, we are use PV62 and PV63 and code names for Status and NewBlock
-      // suggest that there is PV64, but there isn't
+      case (PV64, t) => handlePV64(t, payload)
+
       //common
-      case (_, Status.code63 | Status.code64) => payload.toStatus(msgCode)
-      case (_, SignedTransactions.code) => payload.toSignedTransactions
-      case (_, NewBlock.code63 | NewBlock.code64) => payload.toNewBlock(msgCode)
+      case (_, Codes.SignedTransactionsCode) => payload.toSignedTransactions
+      case (_, Codes.StatusCode) => payload.toStatus
+      case (_, Codes.NewBlockCode) => payload.toNewBlock
 
       case (PV61, t) => handlePV61(t, payload)
 
-      case (PV62 | PV63, pv62.NewBlockHashes.code) => payload.toNewBlockHashes
-      case (PV62 | PV63, pv62.GetBlockHeaders.code) => payload.toGetBlockHeaders
-      case (PV62 | PV63, pv62.BlockHeaders.code) => payload.toBlockHeaders
-      case (PV62 | PV63, pv62.GetBlockBodies.code) => payload.toGetBlockBodies
-      case (PV62 | PV63, pv62.BlockBodies.code) => payload.toBlockBodies
+      case (PV62 | PV63, Codes.NewBlockHashesCode) => payload.toNewBlockHashes
```

Comment:
Shouldn't these messages be handled for PV64 as well? This code is getting a bit out of control so I would suggest to refactor as follows: 

```
def fromBytes(protocolVersion, msgCode, payload) = {
  protocolVersion match {
     case PV64 => handlePV64(msgCode, payload)
     case PV63 => handlePV63(msgCode, payload)
     ...
     case PV60 => handlePV60(msgCode, payload)
     case _ => fail()
  }
}

private def handlePV64(msgCode, payload) = msgCode match {
  case pv64msg1 => decode(payload)
  case pv64msg2 => decode(payload)
  case msgObsoletedByPV64 => fail() // not sure if we need this case for any PV**
  ...
  case _ => handlePV63(msg, payload)
}

private def handlePV63(msgCode, payload) = msgCode match {
  case pv63msg1 => decode(payload)
  case pv63msg2 => decode(payload)
  ...
  case _ => handlePV62(msg, payload)
}

...

private def handlePV60(msgCode, payload) = msgCode match {
  case pv60msg1 => decode(payload)
  case pv60msg2 => decode(payload)
  ...
  case _ => fail()
}
```

This introduces [PV60 as the version for common messages](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#eth60-and-below). Perhaps we should even rename `CommonMessages` to `PV60` as this would give us more consistent naming?

Reply:
In my initial version, Status and NewBlock were part of PV63, but I decided later to move them to CommonMessages again. The true is PV61 Status is different than other ones, but we have not supported it and just use that common Status everywhere besides new PV. I don't know if we want to clean it up or not? @ntallar ? If yes, we can do this in the different PR, as this one is already big one.